waterdata/tests/test_utils.py

```diff
@@ -0,0 +1,100 @@
+                              'water, and meteorological sites\n# for sites in USA.\n#\n# File-format description:  '
+                              'http://help.waterdata.usgs.gov/faq/about-tab-delimited-output\n# Automated-retrieval '
+                              'info: http://waterservices.usgs.gov/rest/Site-Service.html\n#\n# Contact:   '
+                              'gs-w_support_nwisweb@usgs.gov\n#\n# The following selected fields are included in this '
+                              'output:\n#\n#  agency_cd       -- Agency\n'
+                              '#  site_no         -- Site identification number\n#  station_nm      -- Site name\n'
+                              '#  site_tp_cd      -- Site type\n#  dec_lat_va      -- Decimal latitude\n'
+                              '#  dec_long_va     -- Decimal longitude\n'
+                              '#  coord_acy_cd    -- Latitude-longitude accuracy\n'
+                              '#  dec_coord_datum_cd -- Decimal Latitude-longitude datum\n'
+                              '#  alt_va          -- Altitude of Gage/land surface\n'
+                              '#  alt_acy_va      -- Altitude accuracy\n#  alt_datum_cd    -- Altitude datum\n'
+                              '#  huc_cd          -- Hydrologic unit code\n#\nagency_cd\tsite_no\tstation_nm\t'
+                              'site_tp_cd\tdec_lat_va\tdec_long_va\tcoord_acy_cd\tdec_coord_datum_cd\talt_va\t'
+                              'alt_acy_va\talt_datum_cd\thuc_cd\n5s\t15s\t50s\t7s\t16s\t16s\t1s\t10s\t8s\t3s\t10s\t'
+                              '16s\nUSGS\t345670\tSome Random Site\tST\t200.94977778\t-100.12763889\tS\t'
+                              'NAD83\t 151.20\t .1\tNAVD88\t02070010\n')
+        self.test_bad_resp = 'Garbage Text'
+
+    @requests_mock.mock()
+    def test_success(self, r_mock):
+        r_mock.get(self.test_url, status_code=200, text=self.test_rdb_text, reason='OK')
+        result = get_water_services_data(self.test_hostname, 'nwis/site/?site={}'.format(self.test_site_number))
+        expected = (self.test_rdb_text, 200, 'OK')
+        self.assertTupleEqual(expected, result)
+
+    @requests_mock.mock()
+    def test_bad_request(self, r_mock):
+        r_mock.get(self.test_url, status_code=400, text=self.test_bad_resp, reason='Some Reason')
+        result = get_water_services_data(self.test_hostname, 'nwis/site/?site={}'.format(self.test_site_number))
+        expected = (None, 400, 'Some Reason')
+        self.assertTupleEqual(expected, result)
+
+    def test_service_timeout(self):
+        with mock.patch('waterdata.utils.r.get') as r_mock:
+            r_mock.side_effect = r.exceptions.Timeout
+            result = get_water_services_data(self.test_hostname, 'nwis/site/?site={}'.format(self.test_site_number))
+        expected = (None, None, None)
+        self.assertTupleEqual(expected, result)
+
+
+class TestParseRdb(TestCase):
+
+    def setUp(self):
```

Comment:
Should also test an empty string and one that doesn't provide any data lines. Should also have a test that has more than one data row (we will get rdb's like this for site's which are only unique by agency code). For really completeness might want to include a count mismatch between the column header row and a data row.

Reply:
It doesn't really matter but I seem to recall someone saying that we should ignore the more than one site coming back for this version of the view. Considering that there's a separate ticket to deal with that non-unique site codes.

Reply:
It's more a matter of testing a utility function. The function itself should parse any rdb file regardless of whether it's site information, data, or how many rows of data there are.

It should be ignored in terms of not handling it properly in the view.

Reply:
I'm not arguing about the utility of tes,t I'm just pointing out what seems to be lapse in the argument about adding that test. Specifically, there are instances where we haven't over engineered something because it wasn't something we needed at the time if implementation. I would say that:

> Should also have a test that has more than one data row

Within the scope of this ticket is contrary to the arguments where were made during planning.

Reply:
I think this is a more of a testing philosophy issue. A utility function may do more than what you are currently using it for in the code base. But if it was written to handle (in this case), any rdb file, than that's what should be tested, IMO.

Reply:
I suppose one could argue that `parse_rdb` is an unnecessary expansion of scope for what this task  is expected to accomplish.